crates/amaru/src/ledger/store/columns/accounts.rs

```diff
@@ -104,6 +104,29 @@ pub mod rocksdb {
         Ok(())
     }
 
+    /// Reset rewards counter of many accounts.
+    pub fn reset<DB>(
+        db: &Transaction<'_, DB>,
+        rows: impl Iterator<Item = super::Key>,
```

Comment:
Usage of iterator looks a bit fragile to me here as iterator can only be traversed once. This could lead to confusion if some future changes consume the iterator beforehand.

Reply:
My idea behind using an iterator here is double:

1. Avoid committing to a specific data-structure in this interface (since it needs not to)
2. Actually convey the idea that we only need to traverse (once) the data. 

Surely, if the caller partially consume the iterator before passing it then, it's going to be missing entries; but that's a caller's fault / problem (or own desire; as perhaps it is what the caller specifically wants); not an API issue IMO 🤔

Reply:
Makes sense.
Looks like the [rust guideline](https://rust-lang.github.io/api-guidelines/flexibility.html#functions-minimize-assumptions-about-parameters-by-using-generics-c-generic) for functions consuming collections is to use [IntoIterator](https://doc.rust-lang.org/std/iter/trait.IntoIterator.html). Maybe something to consider.